test/RewriteTest/ConditionRegexTest.cs

```diff
@@ -0,0 +1,98 @@
+
+            var expected = new GeneralExpression { Type = ConditionType.Regex, Variable = "(.*)",  Invert = false };
+            Assert.True(CompareConditions(results, expected));
+        }
+
+        [Theory]
+        [InlineData("-d", OperationType.Directory, ConditionType.FileTest)]
+        [InlineData("-f", OperationType.RegularFile, ConditionType.FileTest)]
+        [InlineData("-F", OperationType.ExistingFile, ConditionType.FileTest)]
+        [InlineData("-h", OperationType.SymbolicLink, ConditionType.FileTest)]
+        [InlineData("-L", OperationType.SymbolicLink, ConditionType.FileTest)]
+        [InlineData("-l", OperationType.SymbolicLink, ConditionType.FileTest)]
+        [InlineData("-s", OperationType.Size, ConditionType.FileTest)]
+        [InlineData("-U", OperationType.ExistingUrl, ConditionType.FileTest)]
+        [InlineData("-x", OperationType.Executable, ConditionType.FileTest)]
+        public void ConditionParser_CheckFileOperations(string condition, OperationType operation, ConditionType cond)
+        {
+            var results = ConditionRegexParser.ParseCondition(condition);
+
+            var expected = new GeneralExpression { Type = cond, Operation = operation , Invert = false };
+            Assert.True(CompareConditions(results, expected));
+        }
+
+        [Theory]
+        [InlineData("!-d", OperationType.Directory, ConditionType.FileTest)]
+        [InlineData("!-f", OperationType.RegularFile, ConditionType.FileTest)]
+        [InlineData("!-F", OperationType.ExistingFile, ConditionType.FileTest)]
+        [InlineData("!-h", OperationType.SymbolicLink, ConditionType.FileTest)]
+        [InlineData("!-L", OperationType.SymbolicLink, ConditionType.FileTest)]
+        [InlineData("!-l", OperationType.SymbolicLink, ConditionType.FileTest)]
+        [InlineData("!-s", OperationType.Size, ConditionType.FileTest)]
+        [InlineData("!-U", OperationType.ExistingUrl, ConditionType.FileTest)]
+        [InlineData("!-x", OperationType.Executable, ConditionType.FileTest)]
+        public void ConditionParser_CheckFileOperationsInverted(string condition, OperationType operation, ConditionType cond)
+        {
+            var results = ConditionRegexParser.ParseCondition(condition);
+
+            var expected = new GeneralExpression { Type = cond, Operation = operation, Invert = true };
+            Assert.True(CompareConditions(results, expected));
+        }
+
+        [Theory]
+        [InlineData("-gt1", OperationType.Greater, "1", ConditionType.IntComp)]
+        [InlineData("-lt1", OperationType.Less, "1", ConditionType.IntComp)]
+        [InlineData("-ge1", OperationType.GreaterEqual, "1", ConditionType.IntComp)]
+        [InlineData("-le1", OperationType.LessEqual, "1", ConditionType.IntComp)]
+        [InlineData("-eq1", OperationType.Equal, "1", ConditionType.IntComp)]
+        [InlineData("-ne1", OperationType.NotEqual, "1", ConditionType.IntComp)]
```

Comment:
How about some negative tests that check that your parser handles incorrect expressions?